starters/next-drupal-starter/lib/getPaths.js

```diff
@@ -0,0 +1,75 @@
+import { getCurrentLocaleStore } from "./drupalStateContext";
+/**
+ * @description Helper function to get the current path nedded for getStaticPaths
+ * @param {import('next').GetStaticPathsContext} context - Nextjs getStaticPaths context
+ * @param {import('@pantheon-systems/drupal-kit').DrupalState} globalDrupalStateStores - Drupal state stores from drupalStateContext.js. Can be auth'd stores or non auth'd stores.
+ * @param {string} node - The node to fetch from Drupal. This will be passed to DrupalState.getObject as the objectName.
+ * @param {string} dynamicRouteName - The name of the dynamic route. For example, if using getPath in /pages/articles/[slug] then dynamicRouteName would be "slug".
+ * @param {string} urlAliasPrefix - The prefix of the path alias set in Drupal.
+ * For example, if the alias for articles are prefixed with "/articles" then urlAliasPrefix would be "articles".
+ * @param {boolean} failGracefully - If true, return an empty array which will not render any paths during the build. This is useful when using the fallback option in getStaticPaths.
+ * @see https://nextjs.org/docs/api-reference/data-fetching/get-static-paths for more information on the fallback option.
+ * @returns {Promise<{paths: {params: {dynamicRouteName: string[]}, locale: string}[]}>} an array of paths to return from getStaticPaths.
+ */
+export const getPaths = async (
+  context,
+  globalDrupalStateStores,
+  node,
+  dynamicRouteName,
+  urlAliasPrefix,
+  failGracefully = false
+) => {
+  // Get paths for each locale.
+  const pathsByLocale = context?.locales.map(async (locale) => {
+    const store = getCurrentLocaleStore(locale, globalDrupalStateStores);
+    store.params.clear();
+
+    try {
+      // fetch the node from Drupal
+      const data = await store.getObject({
+        objectName: node,
+        query: `
+            {
+              id
+              path {
+                alias
+              }
+            }
+          `,
+      });
+
+      // map over the data fetch to extract the path name
+      return data.map((datum) => {
+        const regex = new RegExp(`^\/${urlAliasPrefix}\/(.*)$`);
```

Comment:
Trying to think through the cases that this regex won't handle.

* If the file system route is /pages/prefix (for example `/pages/article`) I think deeper routes like `/article/featured/my-article` will still match.

* If the file system route is deeper than one subdir - /pages/prefix/subdir (for example `/pages/article/featured`) I think routes like `/article/featured/my-article` will only work if they use `article` as the prefix, and not `article/featured` because the / needs to be escaped. Maybe we should replace `/` in urlAliasPrefix with `\/`? The alternative of explaining this is comments or docs seems like it would be difficult to do in a way that makes sense.

* I also think that this wouldn't work with paths that aren't prefixed. Some variation on removing the second `\/` in the regex might work - although you might have to make other adjustments due to the fact that the match will have a leading slash I think.

Reply:
```suggestion
         // ^(?:\/?${urlAliasPrefix}\/)? - optional capture group to match leading slash and or the urlAliasPrefix
        // ((?:\/?(.*))$ - capture group that matches path name inside optinal capture group with optional leading slash
        const regex = new RegExp(`^(?:\/?${urlAliasPrefix}\/)?((?:\/?(.*))$`);
```

Thoughts on this possible regex? `^(?:\/?${urlAliasPrefix}\/)?((?:\/?(.*))$`

Reply:
At a glance I think that could work, although the example seems to be missing a closing paren. Instances of `/` would also still have to be handled in the urlAliasPrefix.